lib/livebook/content_loader.ex

```diff
@@ -0,0 +1,93 @@
+defmodule Livebook.ContentLoader do
+  @moduledoc false
+
+  @doc """
+  Rewrite known URLs, so that they point to plain text file rather than HTML.
+
+  Currently the rewerites handle:
+
+    * GitHub files
+    * Gist files
+  """
+  @spec rewrite_url(String.t()) :: String.t()
+  def rewrite_url(url) do
+    url
+    |> URI.parse()
+    |> do_rewrite_url()
+    |> URI.to_string()
+  end
+
+  defp do_rewrite_url(%URI{host: "github.com"} = uri) do
+    case String.split(uri.path, "/") do
+      ["", owner, repo, "blob", hash | file_path] ->
+        path = Enum.join(["", owner, repo, hash | file_path], "/")
+
+        %{
+          uri
+          | path: path,
+            host: "raw.githubusercontent.com",
+            authority: "raw.githubusercontent.com"
+        }
+
+      _ ->
+        uri
+    end
+  end
+
+  defp do_rewrite_url(%URI{host: "gist.github.com"} = uri) do
+    case String.split(uri.path, "/") do
+      ["", owner, hash] ->
+        path = Enum.join(["", owner, hash, "raw"], "/")
```

Comment:
What happens if a gist has multiple files?

Reply:
In that case `/raw` points to the first file. If the user pastes the root Gist URL we cannot determine the specific file, so this is probably best we can do?